src/index.js

```diff
@@ -196,15 +187,10 @@ class JanusAdapter {
     debug("pub waiting for offer");
     var offer = await peerConnection.createOffer();
 
-    debug("pub waiting for ice and descriptions");
-    await Promise.all([
-      iceReady,
-      peerConnection.setLocalDescription(offer),
-      (async () => {
-        const answer = await handle.sendJsep(offer);
-        return peerConnection.setRemoteDescription(answer.jsep);
-      })()
-    ]);
+    debug("pub waiting for local/remote descriptions");
+    await peerConnection.setLocalDescription(offer);
+    const answer = await handle.sendJsep(offer);
+    await peerConnection.setRemoteDescription(answer.jsep);
```

Comment:
These two steps take some tens or hundreds of milliseconds each if I remember correctly, so unless you found they need to be synchronous, I would revert the `await Promise.all(...)`.

Reply:
Oh right. Forgot about that.